test/spec/modules/userId_spec.js

```diff
@@ -1412,4 +1423,191 @@ describe('User ID', function() {
       expect(server.requests[0].url).to.equal('https://match.adsrvr.org/track/rid?ttd_pid=rubicon&fmt=json');
     });
   });
+
+  describe('Consent changes determine getId refreshes', function() {
+    const mockIdCookieName = 'MOCKID';
+    let expStr;
+    let adUnits;
+    let mockGetId = sinon.stub();
+    let mockDecode = sinon.stub();
+    let mockExtendId = sinon.stub();
+    let cmpStub;
+    let testConsentData;
+    const consentConfig = {
+      cmpApi: 'iab',
+      timeout: 7500,
+      allowAuctionWithoutConsent: false
+    };
+    const userIdConfig = {
+      userSync: {
+        userIds: [{
+          name: 'mockId',
+          storage: {
+            name: 'MOCKID',
+            type: 'cookie',
+            refreshInSeconds: 30
+          }
+        }],
+        auctionDelay: 5
+      }
+    };
+    const mockIdSystem = {
+      name: 'mockId',
+      getId: mockGetId,
+      decode: mockDecode,
+      extendId: mockExtendId
+    };
+
+    beforeEach(function () {
+      // clear cookies
+      expStr = (new Date(Date.now() + 25000).toUTCString());
+      coreStorage.setCookie(mockIdCookieName, '', EXPIRED_COOKIE_DATE);
+      coreStorage.setCookie(`${mockIdCookieName}_last`, '', EXPIRED_COOKIE_DATE);
+      coreStorage.setCookie(CONSENT_LOCAL_STORAGE_NAME, '', EXPIRED_COOKIE_DATE);
+
+      // init
+      adUnits = [getAdUnitMock()];
+      init(config);
+
+      // init id system
+      attachIdSystem(mockIdSystem);
+      config.setConfig(userIdConfig);
+
+      // init consent management
```

Comment:
Adding `delete window.__tcfapi;` should fix failing unit tests.

Reply:
ugh...i thought about trying to make sure that was removed, but i thought that since i wasn't checking v2 it was fine. but this definitely works, thanks.

Reply:
i'm actually considering changing the tests to use v2 instead, since that will be what's used longer term. any thoughts? should i support both?

Reply:
i went ahead and added support for tcf 2